nutanix/data_source_nutanix_foundation_central_list_api_keys.go

```diff
@@ -0,0 +1,129 @@
+package nutanix
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-nutanix/client/fc"
+	"github.com/terraform-providers/terraform-provider-nutanix/utils"
+)
+
+func dataSourceNutanixFCListAPIKeys() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceNutanixFCListAPIKeysRead,
+		Schema: map[string]*schema.Schema{
+			"length": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"offset": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"metadata": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"total_matches": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"length": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"offset": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"api_keys": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"alias": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"created_timestamp": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"key_uuid": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"api_key": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"current_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceNutanixFCListAPIKeysRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).FC
+
+	req := &fc.ListMetadataInput{}
+	length, lok := d.GetOk("length")
+	if lok {
+		req.Length = utils.IntPtr(length.(int))
```

Comment:
If length and offset are not given then do we get all the response entities ? 
I guess we don't get there will be one upper limit in case of other APIs we are calling the same API again and again until we get all the entities.

Reply:
yes. we get all responses even we don't give length and offset params.